massa-models/src/denunciation.rs

```diff
@@ -0,0 +1,731 @@
+        };
+
+        let v_endorsement1 =
+            Endorsement::new_verifiable(endorsement_1, EndorsementSerializer::new(), &keypair)
+                .unwrap();
+
+        let endorsement_2 = Endorsement {
+            slot,
+            index: 0,
+            endorsed_block: BlockId(Hash::compute_from("blk2".as_bytes())),
+        };
+
+        let v_endorsement2 =
+            Endorsement::new_verifiable(endorsement_2, EndorsementSerializer::new(), &keypair)
+                .unwrap();
+
+        let endorsement_3 = Endorsement {
+            slot,
+            index: 0,
+            endorsed_block: BlockId(Hash::compute_from("blk3".as_bytes())),
+        };
+        let v_endorsement_3 =
+            Endorsement::new_verifiable(endorsement_3, EndorsementSerializer::new(), &keypair)
+                .unwrap();
+
+        return (
+            slot,
+            keypair,
+            v_endorsement1,
+            v_endorsement2,
+            v_endorsement_3,
+        );
+    }
+
+    #[test]
+    fn test_endorsement_denunciation() {
+        // Create an endorsement denunciation and check if it is valid
+        let (_slot, _keypair, s_endorsement_1, s_endorsement_2, _s_endorsement_3) =
+            gen_endorsements_for_denunciation();
+        let denunciation: Denunciation = (&s_endorsement_1, &s_endorsement_2).try_into().unwrap();
+
+        assert_eq!(denunciation.is_for_endorsement(), true);
+        assert_eq!(denunciation.is_valid().unwrap_or(false), true);
+    }
+
+    #[test]
+    fn test_endorsement_denunciation_invalid_1() {
+        let (slot, keypair, s_endorsement_1, _s_endorsement_2, _s_endorsement_3) =
+            gen_endorsements_for_denunciation();
+
+        // Try to create a denunciation from 2 endorsements @ != index
+        let endorsement_4 = Endorsement {
+            slot,
+            index: 9,
+            endorsed_block: BlockId(Hash::compute_from("foo".as_bytes())),
+        };
+        let s_endorsement_4 =
+            Endorsement::new_verifiable(endorsement_4, EndorsementSerializer::new(), &keypair)
+                .unwrap();
+
+        let denunciation = Denunciation::try_from((&s_endorsement_1, &s_endorsement_4));
+
+        assert!(matches!(denunciation, Err(DenunciationError::InvalidInput)));
+
+        // Try to create a denunciation from only 1 endorsement
+        let denunciation = Denunciation::try_from((&s_endorsement_1, &s_endorsement_1));
+
+        assert!(matches!(denunciation, Err(DenunciationError::InvalidInput)));
+    }
+
+    #[test]
+    fn test_endorsement_denunciation_is_for() {
+        let (slot, keypair, s_endorsement_1, s_endorsement_2, s_endorsement_3) =
+            gen_endorsements_for_denunciation();
+
+        let denunciation: Denunciation = (&s_endorsement_1, &s_endorsement_2).try_into().unwrap();
+
+        assert_eq!(denunciation.is_for_endorsement(), true);
+        assert_eq!(denunciation.is_valid().unwrap_or(false), true);
```

Comment:
In the context of tests, unwrap() should succeed, if it fails it's because we have change something. Here unwarp_or(false) may hide unwanted changes.
You may want to replace by:

```suggestion
        assert_eq!(denunciation.is_valid().unwrap(), true);
```
(also the case in other tests below)

Reply:
done

Reply:
or symply 
```rust
assert!(denunciation.is_valid().unwrap());
```